source/utils.d

```diff
@@ -387,13 +361,84 @@ body
                       url,
                       retryCount,
                       retryCount > 1 ? "times" : "time");
-            downloadFile (url, dest, retryCount - 1);
+            download (url, dest, retryCount - 1);
         } else {
             throw e;
         }
     }
 }
 
+/**
+ * Download or open `path` and return it as a string array.
+ *
+ * Params:
+ *      path = The path to access.
+ *
+ * Returns: The data if successful.
+ */
+string[] getFile (const string path, const uint retryCount = 5)
+{
+    import core.stdc.stdlib : free;
+    import core.sys.linux.stdio : fclose, open_memstream;
+
+    char * ptr = null;
+    scope (exit) free (ptr);
+
+    size_t sz = 0;
+
+    auto f = open_memstream (&ptr, &sz);
```

Comment:
No sorry, I was mainly concerned with how to get a `File` so that I could keep the core of the code the same. You mean just reading into a `string[]` directly I assume?